src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -1,589 +1,131 @@
 package com.eppo.sdk;
 
-import cloud.eppo.ShardUtils;
-import cloud.eppo.Utils;
-import cloud.eppo.rac.Constants;
-import cloud.eppo.rac.dto.*;
-import cloud.eppo.rac.exception.EppoClientIsNotInitializedException;
-import cloud.eppo.rac.exception.InvalidInputException;
-import com.eppo.sdk.helpers.*;
-import com.eppo.sdk.helpers.bandit.BanditEvaluator;
-import com.fasterxml.jackson.databind.JsonNode;
-import java.util.*;
-import java.util.stream.Collectors;
-import org.ehcache.Cache;
+import cloud.eppo.BaseEppoClient;
+import cloud.eppo.logging.AssignmentLogger;
+import cloud.eppo.logging.BanditLogger;
+import com.eppo.sdk.helpers.AppDetails;
+import com.eppo.sdk.helpers.FetchConfigurationsTask;
+import java.util.Timer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class EppoClient {
+public class EppoClient extends BaseEppoClient {
```

Comment:
The way things worked out, I opting for inheritance. My first pass did composition but then `getInstance()` was returning `BaseEppoClient` and there were some protected things (like `loadConfigurations()` I wanted to be able to leverage but not exposed outside of this class. 

It may seem like an odd pattern choice (for now), as the only thing really overridden is the constructor. However it works well and I think it will be more useful for android when we start dealing with thing like caching.